### DISCUSSION TITLE
重定向怎么带上query？

### DISCUSSION THREAD

**Discussion by ayahua** [umijs/umi#9736] [Q&A]
` {
      "path": "/",
      "redirect": "/home"
    }`
路由里设置redirect重定向之后，无法带上原来url上的search内容。然后在onRouteChange方法里，也只能获取重定向之后的路由数据，重定向之前不会触发onRouteChange方法。

**Comment by ayahua** [comment]
又排查了一下，是html设置了base，浏览器从"/"到"/base"跳转的时候，不会触发onRouteChange

**Comment by weilun0510** [comment]
正常跳转怎么带上query?
